src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java

```diff
@@ -288,33 +286,19 @@ void updateContributions(Set<StageWithCaseData> stage) {
     StageWithCaseData getUnassignedAndActiveStageByTeamUUID(UUID teamUUID, UUID userUUID) {
         log.debug("Getting unassigned cases for user: {} in team {}", userUUID, teamUUID);
         Set<StageWithCaseData> unassignedStages = stageRepository.findAllUnassignedAndActiveByTeamUUID(teamUUID);
-        if (unassignedStages.isEmpty()) {
-            log.debug("No unassigned case found for user: {} in team {}", userUUID, teamUUID);
-            return null;
-        }
 
-        for (StageWithCaseData stage : unassignedStages) {
-            stagePriorityCalculator.updatePriority(stage.getData(), stage.getCaseDataType());
-            daysElapsedCalculator.updateDaysElapsed(stage.getData(), stage.getCaseDataType());
-        }
+        unassignedStages.forEach(stage -> stagePriorityCalculator.updatePriority(stage.getData(), stage.getCaseDataType()));
 
-        double prevSystemCalculatedPriority = 0;
-        StageWithCaseData nextAvailableStage = unassignedStages.stream().findFirst().get();
-        for (StageWithCaseData stage : unassignedStages) {
-            JSONObject caseData = new JSONObject(stage.getData());
-            double systemCalculatedPriority = caseData.getDouble("systemCalculatedPriority");
+        var nextAvailableStage = unassignedStages.stream().max(Comparator.comparingDouble(stage -> Double.parseDouble(stage.getData().getOrDefault("systemCalculatedPriority", "0"))));
 
-            if (systemCalculatedPriority > prevSystemCalculatedPriority) {
-                prevSystemCalculatedPriority = systemCalculatedPriority;
-                nextAvailableStage = stage;
-            }
+        if(nextAvailableStage.isPresent()) {
+            StageWithCaseData stage = nextAvailableStage.get();
+            updateStageUser(stage.getCaseUUID(), stage.getUuid(), userUUID);
```

Comment:
Nice catch this to reduce complexity and calculations…

Reply:
again, we were calculating x hundred deadlines, then filtering to first().